### DISCUSSION TITLE
Help creating Regex syntax

### DISCUSSION THREAD

**Discussion by be24horas** [ObsidianToAnki/Obsidian_to_Anki#463] [Q&A]
Hi, I am trying to learn to write Regex expressions to create custom cards.

I started from the Q&A template provided in the GitHub:

^Q: ((?:.+\n)*)\n*A: (.+(?:\n(?:^.{1,3}$|^.{4}(?<!<!--).*))*)

But instead of only "Q" and "A", I needed the following 5 fields: "Fonte" "Tema" "Questão" Gabarito" "Comentário". I then created the following Regex expression:

^Fonte: ((?:.+\n)*)\n*Tema: ((?:.+\n)*)\n*Questão: ((?:.+\n)*)\n*Comentário: ((?:.+\n)*)\n*Gabarito: (.+(?:\n(?:^.{1,3}$|^.{4}(?<!<!--).*))*)

But after the sync between Obsidian and Anki, I only get a card with the content from the first 2 fields "Fonte" and "Tema" 

What am I missing to do in the Regex expression?

**Comment by envico801** [comment]
Hi @be24horas, it is possible that this is happening because you need to create a new note type that has the fields you mention. To do this you need to open Anki and go to `Tools > Manage Note Types > Add > Add: Basic > Give it a name`.

Once you have created the new note type you need to update the plugin data so that it realises that there is a new note type and synchronises it. 

To do this open Obsidian, go left down to the gear (options section), go to the community plugins, click on the Obsidian_to_Anki options and click on the blue button at the bottom of the options that says "Regenerate Note Type Table", to sync the new note types.

![2024-02-15_1-10-13](https://github.com/ObsidianToAnki/Obsidian_to_Anki/assets/132226893/d0829185-6a80-4ecb-bc3b-192ab24903c4)

Once the new note types have been synced you will have to add the custom regex to the new note type you just created.

And then the regex seems to be fine 👍. Each group you capture will represent a field within the note (in order).
